app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,347 @@
+package org.oppia.app.topic
+
+import android.app.Application
+import android.content.Context
+import android.content.res.Configuration
+import androidx.annotation.UiThread
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.Matchers
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPosition
+import org.oppia.app.utility.EspressoTestsMatchers.matchCurrentTabTitle
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for [TopicActivity]. */
+@RunWith(AndroidJUnit4::class)
+class TopicFragmentTest {
+  private val topicName = "Second Test Topic"
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<TopicActivity> = ActivityTestRule(
+    TopicActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Test
+  fun testTopicActivity_clickDummyButton_showsTopicActivityWithMultipleTabs_isTabLayoutDisplayed() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.topic_tabs_container)).perform(click()).check(matches(isDisplayed()))
+    onView(withId(R.id.topic_tabs_container)).perform(click()).check(matches(isDisplayed()))
+
+  }
+
+  @Test
+  fun testTopicActivity_swipePage_hasSwipedPage() {
+    activityTestRule.launchActivity(null)
+    onView(withId(R.id.topic_tabs_viewpager)).check(matches(isDisplayed()))
+    onView(withId(R.id.topic_tabs_viewpager)).perform(ViewActions.swipeLeft())
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_showsTopicActivityWithMultipleTabs_isTabLayoutDisplayedWithOverviewTabSelectedWhenOpenedFirstTime() {
+    activityTestRule.launchActivity(null)
+
+    onView(
+      allOf(
+        withText("OVERVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    )
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("OVERVIEW")))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_isOverviewTabSelectedAndContentMatched() {
```

Comment:
This naming is incorrect:

Suggestion: `testTopicActivity_overviewTabIsSelectedWithCorrectContent_isSuccessful`

Not sure if the suggestion is correct but you should not start and action with `is` in `isOverviewTab...` because that is a form of question.

Update test-case names everywhere.

Reply:
Renamed